From 47a61c5173202e4839e5dbdd2fc23a29235b5e0c Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 30 Jan 2025 14:27:19 +0100
Subject: [PATCH] arm64: dts: st: add rng1 node on stm32mp211

Add support rng1 node for STM32MP21

Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ib5fe9bbf96496ee671685eb0993c07f667868c29
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/433142
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -1627,6 +1627,17 @@
 				status = "disabled";
 			};
 
+			rng1: rng@42030800 {
+				compatible = "st,stm32mp21-rng";
+				reg = <0x42030800 0x0 0x400>;
+				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG1>;
+				clock-names = "rng_clk", "rng_hclk";
+				resets = <&rcc RNG1_R>;
+				access-controllers = <&rifsc 92>;
+				power-domains = <&d1_pd>;
+				status = "disabled";
+			};
+
 			iwdg1: watchdog@44010000 {
 				compatible = "st,stm32mp1-iwdg";
 				reg = <0x44010000 0x0 0x400>;
